### PR TITLE
Add go 1.18 support; set Dockerfile to go 1.18

### PR DESCRIPTION
**Pull request by dragonsinth**
https://github.com/fullstorydev/grpcurl/issues/300

### CODE REVIEW

**Comment by dragonsinth** [comment]
@gpassini @sanjaybv @franklinlindemberg

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,46 +5,38 @@ shared_configs:
...
     docker:
-      - image: circleci/golang:1.16
-    steps:
-      - checkout
-      - restore_cache:
-          keys:
-            - go-mod-v4-{{ checksum "go.sum" }}
-      - run:
-          name: Install Dependencies
-          command: go mod download
```
Comment (gpassini) [inline .circleci/config.yml:30]: We don't need to run this command anymore?

Comment (dragonsinth) [reply .circleci/config.yml:30]: Nah, the necessary modules will be downloaded when it tries to build.  This was broken out before in order to try to populate Go's module cache and then persist it to Circle's cache.  I found this to be a net loss; restoring the cache could take several minutes, while the non-cached versions would completely in like 1 minute.

**Review by gpassini — APPROVED** [review]
LGTM!

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,46 +5,38 @@ shared_configs:
...
-      - run:
-          name: Run tests
-          command: |
-            #mkdir -p /tmp/test-reports
-            #gotestsum --junitfile /tmp/test-reports/unit-tests.xml
-            make ci
-    #- store_test_results:
-    #    path: /tmp/test-reports
+      - image: cimg/go:1.16
+    steps: *full_job_steps
```
Comment (franklinlindemberg) [inline .circleci/config.yml:28]: do you know why only build-1-16 is using full_job_steps?

**Review by franklinlindemberg — APPROVED** [review]
LGTM

Will you do a release with this new version?

File: .circleci/config.yml (hunk 2 above)
Comment (dragonsinth) [reply .circleci/config.yml:28]: I don't; I get the idea of only running the full deal on 1/4 steps, but I'm not sure why Go 1.16 was specifically chosen for this.

**Comment by dragonsinth** [comment]
> Will you do a release with this new version?

Yeah, once I figure out how to do that :D

**Comment by franklinlindemberg** [comment]
@dragonsinth You've probably seem this, but https://github.com/fullstorydev/grpcurl/tree/master/releasing describes the release doc.

I think it expects docker to be running, so we'll need some tweak. Also do you have access to our docker hub? It will be needed to push the image

**Comment by dragonsinth** [comment]
> @dragonsinth You've probably seem this, but https://github.com/fullstorydev/grpcurl/tree/master/releasing describes the release doc.

Yeah, it's a bit of a read tho.  I need to be sure it's up to date.

> I think it expects docker to be running, so we'll need some tweak. Also do you have access to our docker hub? It will be needed to push the image

My plan is to run through it in a Linux VM.  I probably don't have access to docker hub, do you know anything about that?  I'd love to pair up on this-- has anyone other than @jhump ever run through the process before?

**Pull request merged**